src/snowflake/snowpark/session.py

```diff
@@ -590,6 +597,15 @@ def sql_simplifier_enabled(self, value: bool) -> None:
             pass
         self._sql_simplifier_enabled = value
 
+    @cte_optimization_enabled.setter
+    @experimental_parameter(version="1.15.0")
+    def cte_optimization_enabled(self, value: bool) -> None:
```

Comment:
So it's not yet connected with the server side parameter?

Reply:
Yea it seems not in production yet. We can add it in next release